ocs_ci/utility/uninstall_openshift_logging.py

```diff
@@ -0,0 +1,53 @@
+"""
+Function to teardown the openshift-logging
+"""
+import logging
+
+from ocs_ci.ocs import constants, ocp
+from ocs_ci.ocs.resources.pod import get_all_pods, verify_pod_exists
+
+logger = logging.getLogger(__name__)
+
+
+def uninstall_cluster_logging():
+    """
+    Function to uninstall cluster-logging from the cluster
+    1. Deletes the project "openshift-logging"
+        & "openshift-operators-redhat"
+    2. Deletes the CSC resource "cluster-logging-operator"
+        & "elasticsearch"
+    """
+    # Validating the pods before deleting the instance
+    pod_list = get_all_pods(namespace='openshift-logging')
+    for pod in pod_list:
+        logger.info(f"Pods running in the openshift-logging namespace {pod.name}")
+
+    # Deleting the clusterlogging instance
+    clusterlogging_obj = ocp.OCP(
+        kind='ClusterLogging', namespace='openshift-logging'
+    )
+    assert clusterlogging_obj.delete(resource_name='instance')
+
+    # Verifying the pods if exists after deleting instance
+    pod_list = [
```

Comment:
pod_list is still the same list as before the deletion.
you'll need to call get_all_pods() again to get a fresh list

Reply:
I checked it. Initially the pod list contained 11 items including cluster-logging-operator, after the removal of one item from the list it became 10 excluding cluster-logging-operator. Attaching the rough logs for the reference
```17:50:26 - MainThread - ocs_ci.utility.utils - INFO - Executing command: oc -n openshift-logging --kubeconfig /home/sraghave/deploy-xlarge/auth/kubeconfig get Pod  -o yaml
17:50:29 - MainThread - ocs_ci.utility.uninstall_openshift_logging - INFO - 11
17:50:29 - MainThread - ocs_ci.utility.uninstall_openshift_logging - INFO - Pods running in the openshift-logging namespace cluster-logging-operator-dd48849c-dn4rw
17:50:29 - MainThread - ocs_ci.utility.uninstall_openshift_logging - INFO - Pods running in the openshift-logging namespace elasticsearch-cdm-p3t4ynyx-1-55f577bb95-5x2zc
17:50:29 - MainThread - ocs_ci.utility.uninstall_openshift_logging - INFO - Pods running in the openshift-logging namespace elasticsearch-cdm-p3t4ynyx-2-65b4d4d4fb-42njl
17:50:29 - MainThread - ocs_ci.utility.uninstall_openshift_logging - INFO - Pods running in the openshift-logging namespace elasticsearch-cdm-p3t4ynyx-3-58fd8b7546-zl9zn
17:50:29 - MainThread - ocs_ci.utility.uninstall_openshift_logging - INFO - Pods running in the openshift-logging namespace fluentd-5nw8c
17:50:29 - MainThread - ocs_ci.utility.uninstall_openshift_logging - INFO - Pods running in the openshift-logging namespace fluentd-drz85
17:50:29 - MainThread - ocs_ci.utility.uninstall_openshift_logging - INFO - Pods running in the openshift-logging namespace fluentd-gptkt
17:50:29 - MainThread - ocs_ci.utility.uninstall_openshift_logging - INFO - Pods running in the openshift-logging namespace fluentd-jnfms
17:50:29 - MainThread - ocs_ci.utility.uninstall_openshift_logging - INFO - Pods running in the openshift-logging namespace fluentd-xg2b6
17:50:29 - MainThread - ocs_ci.utility.uninstall_openshift_logging - INFO - Pods running in the openshift-logging namespace fluentd-z6c5x
17:50:29 - MainThread - ocs_ci.utility.uninstall_openshift_logging - INFO - Pods running in the openshift-logging namespace kibana-5bd5544f87-vrssw
17:50:29 - MainThread - ocs_ci.utility.uninstall_openshift_logging - INFO - 10
17:50:29 - MainThread - ocs_ci.utility.utils - INFO - Executing command: oc -n openshift-logging --kubeconfig /home/sraghave/deploy-xlarge/auth/kubeconfig get Pod  -o yaml
17:50:32 - MainThread - ocs_ci.ocs.resources.pod - INFO - Pod elasticsearch-cdm-p3t4ynyx-1-55f577bb95-5x2zc exists in the given namespace
17:50:32 - MainThread - ocs_ci.utility.utils - INFO - Executing command: oc -n openshift-logging --kubeconfig /home/sraghave/deploy-xlarge/auth/kubeconfig get Pod  -o yaml
17:50:35 - MainThread - ocs_ci.ocs.resources.pod - INFO - Pod elasticsearch-cdm-p3t4ynyx-2-65b4d4d4fb-42njl exists in the given namespace
17:50:35 - MainThread - ocs_ci.utility.utils - INFO - Executing command: oc -n openshift-logging --kubeconfig /home/sraghave/deploy-xlarge/auth/kubeconfig get Pod  -o yaml
17:50:38 - MainThread - ocs_ci.ocs.resources.pod - INFO - Pod elasticsearch-cdm-p3t4ynyx-3-58fd8b7546-zl9zn exists in the given namespace
17:50:38 - MainThread - ocs_ci.utility.utils - INFO - Executing command: oc -n openshift-logging --kubeconfig /home/sraghave/deploy-xlarge/auth/kubeconfig get Pod  -o yaml
17:50:41 - MainThread - ocs_ci.ocs.resources.pod - INFO - Pod fluentd-5nw8c exists in the given namespace
17:50:41 - MainThread - ocs_ci.utility.utils - INFO - Executing command: oc -n openshift-logging --kubeconfig /home/sraghave/deploy-xlarge/auth/kubeconfig get Pod  -o yaml
17:50:43 - MainThread - ocs_ci.ocs.resources.pod - INFO - Pod fluentd-drz85 exists in the given namespace
17:50:43 - MainThread - ocs_ci.utility.utils - INFO - Executing command: oc -n openshift-logging --kubeconfig /home/sraghave/deploy-xlarge/auth/kubeconfig get Pod  -o yaml
17:50:46 - MainThread - ocs_ci.ocs.resources.pod - INFO - Pod fluentd-gptkt exists in the given namespace
17:50:46 - MainThread - ocs_ci.utility.utils - INFO - Executing command: oc -n openshift-logging --kubeconfig /home/sraghave/deploy-xlarge/auth/kubeconfig get Pod  -o yaml
17:50:49 - MainThread - ocs_ci.ocs.resources.pod - INFO - Pod fluentd-jnfms exists in the given namespace
17:50:49 - MainThread - ocs_ci.utility.utils - INFO - Executing command: oc -n openshift-logging --kubeconfig /home/sraghave/deploy-xlarge/auth/kubeconfig get Pod  -o yaml
17:50:52 - MainThread - ocs_ci.ocs.resources.pod - INFO - Pod fluentd-xg2b6 exists in the given namespace
17:50:52 - MainThread - ocs_ci.utility.utils - INFO - Executing command: oc -n openshift-logging --kubeconfig /home/sraghave/deploy-xlarge/auth/kubeconfig get Pod  -o yaml
17:50:54 - MainThread - ocs_ci.ocs.resources.pod - INFO - Pod fluentd-z6c5x exists in the given namespace
17:50:54 - MainThread - ocs_ci.utility.utils - INFO - Executing command: oc -n openshift-logging --kubeconfig /home/sraghave/deploy-xlarge/auth/kubeconfig get Pod  -o yaml
17:50:57 - MainThread - ocs_ci.ocs.resources.pod - INFO - Pod kibana-5bd5544f87-vrssw exists in the given namespace
17:50:57 - MainThread - ocs_ci.utility.utils - INFO - Executing command: oc -n openshift-logging --kubeconfig /home/sraghave/deploy-xlarge/auth/kubeconfig get Namespaces  -o yaml
17:50:59 - MainThread - ocs_ci.utility.uninstall_openshift_logging - INFO - The projects got deleted successfully```

Reply:
from a pythonic POV that not possible.
I'm not sure how that's possible that pod_list was changed bug please call get_all_pods again to refresh the list

Reply:
@RazTamir Done.